internal/pkg/term/selector/cfn.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package selector
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/copilot-cli/internal/pkg/template"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"gopkg.in/yaml.v3"
+)
+
+// CFNSelector represents a selector for a CloudFormation template.
+type CFNSelector struct {
+	prompt Prompter
+}
+
+// NewCFNSelector initializes a CFNSelector.
+func NewCFNSelector(prompt Prompter) *CFNSelector {
+	return &CFNSelector{
+		prompt: prompt,
+	}
+}
+
+// Resources prompts the user to multiselect resources from a CloudFormation template body.
+// By default, the prompt filters out any custom resource in the template.
+func (sel *CFNSelector) Resources(msg, finalMsg, help, body string) ([]template.CFNResource, error) {
+	tpl := struct {
+		Resources map[string]struct {
+			Type string `yaml:"Type"`
+		} `yaml:"Resources"`
+	}{}
+	if err := yaml.Unmarshal([]byte(body), &tpl); err != nil {
+		return nil, fmt.Errorf("unmarshal CloudFormation template: %v", err)
+	}
+
+	// Prompt for a selection.
+	var options []prompt.Option
+	for name, resource := range tpl.Resources {
+		if resource.Type == "AWS::Lambda::Function" || strings.HasPrefix(resource.Type, "Custom::") {
```

Comment:
Maybe it makes more sense to filter out from a specific list but I guess it's ok to be more restrictive at first.

Reply:
I kept debating if this method should take as input something like:
```go
func Resources(body string, filterFn ...func(template.CFNResource) bool) ([]template.CFNResource, error)
```
Happy to change it to that boss, if you prefer it to! It's definitely more flexible 👍 at the end I just kept it a bit simpler for now

Reply:
Yeah i think we can keep things simple for now